### DISCUSSION TITLE
How to integrate own negative edges in TGN example?

### DISCUSSION THREAD

**Discussion by ArchitNK** [pyg-team/pytorch_geometric#8111] [Q&A] [answered]
Hello, Thank you for the great library and support for Graphs.
I am working on a dataset that requires temporal context. So I am using [TGN example](https://github.com/pyg-team/pytorch_geometric/blob/master/examples/tgn.py) from the PyG library.
1) I have a dataset where I have some interactions where there is no edge (Negative edges). 
I want to integrate them into training instead of the artificial Negative edge sampling from dataloader. 
In short I want my to define my own neg_dst. Is there any way to do that?

2) The current [neg edge sampling](https://pytorch-geometric.readthedocs.io/en/latest/_modules/torch_geometric/loader/temporal_dataloader.html) generates random sample nodes where min(original nodes)< **node** < max(original node). This could create nodes that are not there at all in the original nodes.
Eg. Original nodes[1,2,5] -> negative sampling could give [1,2,**3,4**,5]
Is there any way to ensure that with negative edge sampling only nodes from original list would be created?

**Comment by rusty1s** [comment, marked as answer]
You can consider overwriting [`TemporalDataLoader`](https://github.com/pyg-team/pytorch_geometric/blob/master/torch_geometric/loader/temporal_dataloader.py) to implement your own negative sampling strategy. Would that work for you?

**Reply by ArchitNK** [reply]
Yes, That could work. Just wanted to know if there is already a method that can do this within PyG

**Reply by rusty1s** [reply]
Currently no, this would be the recommended way for now.

**Reply by ArchitNK** [reply]
Thank you for the suggestion.
